src/flamenco/runtime/fd_bank.c

```diff
@@ -445,6 +445,7 @@ fd_banks_get_bank( fd_banks_t * banks, ulong slot ) {
   bank = fd_banks_map_ele_query( bank_map, &slot, NULL, bank_pool );
   if( FD_UNLIKELY( !bank ) ) {
     FD_LOG_WARNING(( "Failed to get bank" ));
+    fd_rwlock_unread( &banks->rwlock );
```

Comment:
wait shouldnt the caller be retaining the read lock here? how does the pointer stay valid on return -- its not a copy?

Reply:
github's PR UI is idiotic and doesn't let me tag other lines, but referring to line 452 not this